ml-agents/mlagents/trainers/models.py

```diff
@@ -229,28 +230,101 @@ def create_visual_observation_encoder(
         :param h_size: Hidden layer size.
         :param activation: What type of activation function to use for layers.
         :param num_layers: number of hidden layers to create.
+        :param encoder_type: type of visual encoder (default/nature_cnn/resnet)
         :return: List of hidden layer tensors.
         """
         with tf.variable_scope(scope):
-            conv1 = tf.layers.conv2d(
-                image_input,
-                16,
-                kernel_size=[8, 8],
-                strides=[4, 4],
-                activation=tf.nn.elu,
-                reuse=reuse,
-                name="conv_1",
-            )
-            conv2 = tf.layers.conv2d(
-                conv1,
-                32,
-                kernel_size=[4, 4],
-                strides=[2, 2],
-                activation=tf.nn.elu,
-                reuse=reuse,
-                name="conv_2",
-            )
-            hidden = c_layers.flatten(conv2)
+            if encoder_type == "resnet":
```

Comment:
Can we add an enum for these instead? https://docs.python.org/3/library/enum.html

Reply:
added